src/csvcubed/readers/cubeconfig/v1/columnschema.py

```diff
@@ -240,43 +240,72 @@ class AttributeValue(SchemaBaseClass):
 
 
 @dataclass
-class ExistingAttribute(SchemaBaseClass):
+class ExistingAttributeLiteral(SchemaBaseClass):
     from_existing: str
-    data_type: Optional[str] = None
+    data_type: str
     required: bool = False
+
+    def map_to_existing_qb_attribute(self) -> ExistingQbAttributeLiteral:
+
+        _logger.debug(f"Identified {self.as_dict()} as ExistingAttributeLiteral")
+
+        return ExistingQbAttributeLiteral(
+            attribute_uri=self.from_existing,
+            is_required=self.required,
+            data_type=self.data_type,
+        )
+
+
+@dataclass
+class ExistingAttributeResource(SchemaBaseClass):
+    from_existing: str
     values: Union[bool, List[AttributeValue]] = True
-    cell_uri_template: Optional[str] = None
+    required: bool = False
 
     def map_to_existing_qb_attribute(
         self, data: PandasDataTypes
-    ) -> Union[ExistingQbAttribute, ExistingQbAttributeLiteral]:
+    ) -> ExistingQbAttribute:
 
-        if self.data_type is None:
-            return ExistingQbAttribute(
-                self.from_existing,
-                new_attribute_values=_get_new_attribute_values(data, self.values),
-                is_required=self.required,
-            )
+        _logger.debug(f"Identified {self.as_dict()} as ExistingAttributeResource")
 
-        else:
-            if self.values:
-                raise Exception(
-                    "Attributes cannot represent both literal values and attribute (resource) values"
-                )
-            return ExistingQbAttributeLiteral(
-                attribute_uri=self.from_existing,
-                data_type=self.data_type,
-                is_required=self.required,
-            )
+        return ExistingQbAttribute(
+            self.from_existing,
+            new_attribute_values=_get_new_attribute_values(data, self.values),
+            is_required=self.required,
+        )
 
 
 @dataclass
-class NewAttribute(SchemaBaseClass):
+class NewAttributeLiteral(SchemaBaseClass):
+    label: str
+    data_type: str
+    description: Optional[str] = None
+    from_existing: Optional[str] = None
+    definition_uri: Optional[str] = None
+    required: bool = False
+    cell_uri_template: Optional[str] = None
```

Comment:
Literal values (numbers, strings, etc.) don't have URIs so it should not be possible to specify a `cell_uri_template` here.